cpp/core/shuffle/ShuffleWriter.h

```diff
@@ -288,17 +257,17 @@ class ShuffleWriter {
       : numPartitions_(numPartitions),
         partitionWriterCreator_(std::move(partitionWriterCreator)),
         options_(std::move(options)),
-        pool_(std::make_shared<ShuffleMemoryPool>(options_.memory_pool)),
+        splitBufferPool_(std::make_shared<ShuffleMemoryPool>(options_.memory_pool)),
```

Comment:
changing the type of `splitBufferPool_` to `ShuffleMemoryPool` may be better.

the lifecycle of `splitBufferPool_` is binding with ShuffleWriter.